shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -238,6 +219,48 @@ FlutterWindowsEngine::FlutterWindowsEngine(
       std::make_unique<PlatformHandler>(messenger_wrapper_.get(), this);
   settings_plugin_ = std::make_unique<SettingsPlugin>(messenger_wrapper_.get(),
                                                       task_runner_.get());
+
+  // Makeshift PV channel before a separate plugin is there.
+  platform_view_channel_ = std::make_unique<MethodChannel<EncodableValue>>(
+      messenger_wrapper_.get(), "flutter/platform_views",
+      &StandardMethodCodec::GetInstance());
+  platform_view_channel_->SetMethodCallHandler(
+      [this](const MethodCall<EncodableValue>& call,
+             std::unique_ptr<MethodResult<EncodableValue>> result) {
+        const std::string& name = call.method_name();
+        if (name == "create") {
+          FML_LOG(ERROR) << "Receiving creation method";
+          const auto& args = std::get<EncodableMap>(*call.arguments());
+          auto id = std::get<int32_t>(args.find(EncodableValue("id"))->second);
+          FML_LOG(ERROR) << "Found ID for platform view: " << id;
+          auto type = std::get<std::string>(
+              args.find(EncodableValue("viewType"))->second);
+          FML_LOG(ERROR) << "Found view type for platform view: " << type;
+          PlatformViewCreationParams creation_args{view_->GetPlatformWindow(),
+                                                   id, type.data()};
+          CreatePlatformView(creation_args);
+          result->Success();
+        } else if (name == "focus") {
+          const auto& args = std::get<EncodableMap>(*call.arguments());
+          auto id = std::get<int32_t>(args.find(EncodableValue("id"))->second);
+          auto focus =
+              std::get<bool>(args.find(EncodableValue("focus"))->second);
+          pv_reason_ = std::get<int>(args.find(EncodableValue("dir"))->second);
```

Comment:
In order for the platform view to detect from which direction it gained focus, it can query this value from the engine. Alternatively, we could allow/require users to provide a function callback for each platform view type that is called when focus is transferred to the platform view via keyboard navigation, which accepts the direction as an argument.